c2cgeoportal/views/layers.py

```diff
@@ -283,8 +293,27 @@ def security_cb(r, feature, o):
             if allowed.scalar() == 0:
                 raise HTTPForbidden()
 
-        protocol = self._get_protocol_for_layer(layer, before_update=security_cb)
-        return protocol.update(self.request, feature_id)
+            # check is geometry is valid
+            self.validate_geometry_(spatial_elt)
+
+        protocol = self._get_protocol_for_layer(layer, before_update=check_geometry)
+        try:
+            feature = protocol.update(self.request, feature_id)
+            return feature
+        except TopologicalError, e:
+            self.request.response.status_int = 400
+            return {'validation_error': str(e)}
+
+    def validate_geometry_(self, geom):
```

Comment:
Why this name ?
To mark it as private it should be prefixed by an underscore, not postfixed ...
https://docs.python.org/2/tutorial/classes.html#private-variables-and-class-local-references